src/main/java/de/craftlancer/clcapture/CapturePoint.java

```diff
@@ -222,22 +188,83 @@ private void runInactive() {
+        //See if there are multiple clans by comparing clans
+        areMultipleClans = checkAreMultipleClans(inRegionList, areMultipleClans);
         
-        if (tickId % 20 == 0) {
-            bar.getPlayers().stream().filter(a -> a.getWorld().equals(signLocation.getWorld()))
-               .filter(a -> a.getLocation().distance(signLocation) >= type.getBossbarDistance()).forEach(bar::removePlayer);
-            Bukkit.getOnlinePlayers().stream().filter(a -> a.getWorld().equals(signLocation.getWorld()))
-                  .filter(a -> a.getLocation().distance(signLocation) < type.getBossbarDistance()).forEach(bar::addPlayer);
+        //If there are multiple clans present in the region, don't add to bossbar/map
+        if (!areMultipleClans) {
+            scoreMultiplier = handleSingleClanVariables(inRegionList, scoreMultiplier);
+            handleBossBarAndTime(scoreMultiplier);
         }
         
-        if (timeMap.getOrDefault(currentOwner, 0) >= type.getCaptureTime())
+        if (currentOwner == null)
+            setClanColors(null);
+        
+        //If there is a winner, handle the win and return
+        if (timeMap.getOrDefault(currentOwner, 0) >= type.getCaptureTime()) {
             handleWin();
+            return;
+        }
+        
+        //Set variables back to default values
+        previousOwner = currentOwner;
+        currentOwner = null;
+    }
+    
+    //If there are people in the region, set the appropriate variables
+    private int handleSingleClanVariables(List<UUID> inRegionList, int scoreMultiplier) {
+        if (inRegionList.size() > 0) {
+            currentOwner = inRegionList.get(0);
+            if (previousMessageOwner == null || previousMessageOwner != currentOwner) {
+                previousMessageOwner = currentOwner;
+                announce();
+            }
+            if (currentOwner != previousOwner)
+                setClanColors(plugin.getClanPlugin().getClanByUUID(currentOwner));
+            scoreMultiplier = inRegionList.size();
+            timeMap.putIfAbsent(currentOwner, 0);
+        }
+        return scoreMultiplier;
+    }
+    
+    
+    private boolean checkAreMultipleClans(List<UUID> inRegionList, boolean areMultipleClans) {
+        for (UUID a : inRegionList)
+            for (UUID b : inRegionList)
+                if (!a.equals(b)) {
+                    return areMultipleClans;
+                }
+        return false;
+    }
+    
+    private void handleBossBarAndTime(int scoreMultiplier) {
+        //If the owner isn't standing on it, the time will dwindle down
+        timeMap.replaceAll((a, b) -> a.equals(currentOwner) ? b + scoreMultiplier : Math.max(b - scoreMultiplier, 0));
+        //To prevent numbers over 1, check was put in place
+        double progress = timeMap.getOrDefault(currentOwner, 0) / (double) type.getCaptureTime();
+        if (progress <= 1)
+            bar.setProgress(progress);

```

Comment:
Maybe change to `bar.setProgress(Math.min(progress, 1D))`. Does about the same, but more concise.